flax/linen/module.py

```diff
@@ -1488,12 +1488,61 @@ def init(self,
            **kwargs) -> FrozenVariableDict:
     """Initializes a module method with variables and returns modified variables.
 
+    ``init`` takes a ``PRNGKey`` as its first argument and all the ``*args`` and ``**kwargs``
+    that the module's ``__call__`` method takes, and returns a dictionary of initialized
+    variables.
+
+    Example::
+
+      >>> import flax.linen as nn
+      >>> import jax.numpy as jnp
+      >>> import jax
+      ...
+      >>> class Foo(nn.Module):
+      ...   @nn.compact
+      ...   def __call__(self, x, train):
+      ...     x = nn.Dense(16)(x)
+      ...     x = nn.BatchNorm(use_running_average=not train)(x)
+      ...     x = nn.relu(x)
+      ...     return nn.Dense(1)(x)
+      ...
+      >>> module = Foo()
+      >>> key = jax.random.PRNGKey(0)
+      >>> variables = module.init(key, jnp.empty((1, 7)), train=False)
+
+    By default the ``PRNGKey`` will only be used to feed the ``'params'`` RNG stream. If you
+    want to use a different RNG stream or need to use multiple streams, you must pass a
+    dictionary of ``PRNGKey`` to ``init``.
+
+    Example::
+
+      >>> class Foo(nn.Module):
+      ...   @nn.compact
+      ...   def __call__(self, x, train):
+      ...     x = nn.Dense(16)(x)
+      ...     x = nn.BatchNorm(use_running_average=not train)(x)
+      ...     x = nn.relu(x)
+      ...
+      ...     # Add gaussian noise
+      ...     noise_key = self.make_rng('noise')
+      ...     x = x + jax.random.normal(noise_key, x.shape)
+      ...
+      ...     return nn.Dense(1)(x)
+      ...
+      >>> module = Foo()
+      >>> rngs = {'params': jax.random.PRNGKey(0), 'noise': jax.random.PRNGKey(1)}
+      >>> variables = module.init(rngs, jnp.empty((1, 7)), train=False)
+
```

Comment:
Perhaps say that we strongly recommend always jitting init. Actually maybe we should even throw a warning if someone calls init without jitting?

Reply:
Not sure we should strongly recommend it if we are going to implement JAXPR based init in the future.

Reply:
Hmm yes that is true. Then I think we should refer to the jaxpr-based init function from here when it is there.